packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -612,17 +612,17 @@ abstract class FlutterCommand extends Command<void> {
       valueHelp: 'foo=bar',
       splitCommas: false,
     );
-    useDartDefineConfigJsonFileOption();
+    useDartDefineFromFileOption();
   }
 
-  void useDartDefineConfigJsonFileOption() {
+  void useDartDefineFromFileOption() {
     argParser.addMultiOption(
       FlutterOptions.kDartDefineFromFileOption,
-      help: 'The path of a json format file where flutter define a global constant pool. '
-          'Json entry will be available as constants from the String.fromEnvironment, bool.fromEnvironment, '
-          'and int.fromEnvironment constructors; the key and field are json values.\n'
+      help: 'The path of a json format file or .env file where flutter define a global constant pool. '
+          'File entries will be available as constants from the String.fromEnvironment, bool.fromEnvironment, '
+          'and int.fromEnvironment constructors; the key and field can be json values or .env values.\n'
```

Comment:
Consider making this more concise and simple.

While we are here, I think we should remove the bit about `global constant pool`. I think "environment variables" are a more widely-recognized concept, and this terminology would be more precise here.

```suggestion
      help: 'The path of a .json or .env file containing key-value pairs that will be available as environment variables.\n'
            'These can be accessed using the String.fromEnvironment, bool.fromEnvironment, and int.fromEnvironment constructors.',
```